source/pdo_sqlsrv/pdo_stmt.cpp

```diff
@@ -401,9 +401,12 @@ int pdo_sqlsrv_stmt_close_cursor(pdo_stmt_t *stmt TSRMLS_DC)
 
         // to "close the cursor" means we make the statement ready for execution again.  To do this, we 
         // skip all the result sets on the current statement.
-        while( driver_stmt->past_next_result_end == false ) {
-
-            core_sqlsrv_next_result( driver_stmt TSRMLS_CC );
+        // If the statement has not been executed there are no next results to iterate over.
+        if ( driver_stmt->executed == true )
+        {
+            while( driver_stmt->past_next_result_end == false ) {
+                core_sqlsrv_next_result( driver_stmt TSRMLS_CC );
```

Comment:
Sure. But as stated in the documentation looping the cursor until the end is only the "fallback" solution if no more efficient approach is available. Since looking at [sqlsrv_cancel](https://github.com/Microsoft/msphpsql/blob/a02691e277d5804c3d8b28fcdfee31e6fd2df8ab/source/sqlsrv/stmt.cpp#L1308-L1335) seems to utilize an optimized native solution, couldn't that fit in here, too? That looks better than looping through until the end of the cursor is reached.